spec/index.bs

```diff
@@ -1799,20 +1799,90 @@ dictionary GPUBindGroupLayoutEntry {
     required GPUIndex32 binding;
     required GPUShaderStageFlags visibility;
     required GPUBindingType type;
-    GPUTextureViewDimension viewDimension = "2d";
-    GPUTextureComponentType textureComponentType = "float";
-    GPUTextureFormat storageTextureFormat;
-    boolean multisampled = false;
+
+    // Used for uniform buffer and storage buffer bindings.
     boolean hasDynamicOffset = false;
+
+    // Used for sampled texture and storage texture bindings.
+    GPUTextureViewDimension viewDimension;
+
+    // Used for sampled texture bindings.
+    GPUTextureComponentType textureComponentType;
+    boolean multisampled = false;
+
+    // Used for storage texture bindings.
+    GPUTextureFormat storageTextureFormat;
 };
 </script>
 
-  * {{GPUBindGroupLayoutEntry/binding}}:
-    A unique identifier for a resource binding within a {{GPUBindGroupLayoutEntry}}, a corresponding {{GPUBindGroupEntry}}, and shader stages.
+<dl dfn-type=dict-member dfn-for=GPUBindGroupLayoutEntry>
+    : <dfn>binding</dfn>
+    ::
+        A unique identifier for a resource binding within a
+        {{GPUBindGroupLayoutEntry}}, a corresponding {{GPUBindGroupEntry}},
+        and shader stages.
+
+    : <dfn>visibility</dfn>
+    ::
+        A bitset of the members of {{GPUShaderStage}}.
+        Each set bit indicates that a {{GPUBindGroupLayoutEntry}}'s resource
+        will be accessible from the associated shader stage.
+
+    : <dfn>hasDynamicOffset</dfn>
+    ::
+        If the {{GPUBindGroupLayoutEntry/type}} is
+        {{GPUBindingType/"uniform-buffer"}},
+        {{GPUBindingType/"storage-buffer"}}, or
+        {{GPUBindingType/"readonly-storage-buffer"}},
+        this indicates whether a binding requires a dynamic offset.
+
+        Otherwise, it must be `false`.
+
+    : <dfn>viewDimension</dfn>
+    ::
+        If the {{GPUBindGroupLayoutEntry/type}} is
+        {{GPUBindingType/"sampled-texture"}},
+        {{GPUBindingType/"readonly-storage-texture"}}, or
+        {{GPUBindingType/"writeonly-storage-texture"}},
+        this defaults to {{GPUTextureViewDimension/"2d"}}
```

Comment:
I don't understand how this mechanism of "defaults to" works. The user leaves it undefined, but we trieat it as something else?

Reply:
Yes. This is just moving the previous dictionary member default values out of the IDL and into this prose.